### PR TITLE
Add some utility functions to support finalizers

### PR DESCRIPTION
**Pull request by rramkumar1**
Adding some functions needed to support finalizers. This code can easily be reused by other modules.

Much of this code was adapted from https://github.com/kubernetes/kubernetes/pull/54569

/assign @bowei @MrHohn

### CODE REVIEW

File: pkg/utils/finalizer.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package utils
```
Comment (MrHohn) [inline pkg/utils/finalizer.go]: Need copyright boilerplate here.

File: pkg/utils/finalizer_test.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
+package utils
```
Comment (MrHohn) [inline pkg/utils/finalizer_test.go]: Need copyright boilerplate here.

Comment (rramkumar1) [reply pkg/utils/finalizer_test.go]: Done

File: pkg/utils/finalizer.go (hunk 1 above)
Comment (rramkumar1) [reply pkg/utils/finalizer.go]: Done

**Pull request merged**